src/azure-cli/azure/cli/command_modules/resource/_utils.py

```diff
@@ -0,0 +1,51 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import re
+
+_subscription_pattern = r"^\/?subscriptions\/(?P<subscription_id>[a-f0-9-]+)"
+_resource_group_pattern = r"^\/resourceGroups\/(?P<resource_group_name>[-\w\._\(\)]+)"
+_relative_resource_id_pattern = r"^\/providers/(?P<relative_resource_id>.+$)"
+
+
+def parse_resource_id(resource_id):
```

Comment:
Is it possible to utilize  parse_resource_id here?
`from msrestazure.tools import parse_resource_id`

Reply:
Just learnt there's already an existing one. However it may not be suitable to use it here, because it returns a dictionary with unpredictable number of keys, which is dependent on how many child resource namespaces a resource ID has. What I need is just to split a resource id into two parts, the resource scope, and the rest (relative resource ID). If using the existing function I have to process the result dictionary carefully to construct the relative resource ID with the key/value pairs in the result dictionary, which seems to be a harder task. I'm open to changing the utility function's name to `split_resource_id` if that you are okay with that.

Reply:
I see